src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitService.kt

```diff
@@ -144,33 +132,57 @@ class VisitService(
       "visit-scheduler-prison-visit-updated",
       listOfNotNull(
         "reference" to reference,
-        if (updateVisitRequest.prisonerId != null) "prisonerId" to updateVisitRequest.prisonerId else null,
-        if (updateVisitRequest.prisonId != null) "prisonId" to updateVisitRequest.prisonId else null,
-        if (updateVisitRequest.visitType != null) "visitType" to updateVisitRequest.visitType.name else null,
-        if (updateVisitRequest.visitRoom != null) "visitRoom" to updateVisitRequest.visitRoom else null,
+        "prisonerId" to visitEntity.prisonerId,
+        "prisonId" to visitEntity.prisonId,
+        "visitType" to visitEntity.visitType.name,
+        "visitRoom" to visitEntity.visitRoom,
         if (updateVisitRequest.visitRestriction != null) "visitRestriction" to updateVisitRequest.visitRestriction.name else null,
         if (updateVisitRequest.startTimestamp != null) "visitStart" to updateVisitRequest.startTimestamp.format(DateTimeFormatter.ISO_DATE_TIME) else null,
-        if (updateVisitRequest.visitStatus != null) "visitStatus" to updateVisitRequest.visitStatus.name else null
+        if (visitEntity.visitStatus != null) "visitStatus" to visitEntity.visitStatus.name else null
       ).toMap(),
       null
     )
 
     return VisitDto(visitEntity)
   }
 
-  fun deleteVisit(reference: String) {
-    val visit = visitRepository.findByReference(reference)
+  fun bookVisit(reference: String): VisitDto {
 
-    visit?.let {
-      visitRepository.delete(it)
-      telemetryClient.trackEvent(
-        "visit-scheduler-prison-visit-deleted",
-        mapOf(
-          "reference" to reference
-        ),
-        null
-      )
-    } ?: run { log.debug("Visit reference $reference not found") }
+    var revisedVisit = false
+    val existingBookedVisit = visitRepository.findBookedVisit(reference)
+    existingBookedVisit?.let {
+      existingBookedVisit.visitStatus = VisitStatus.CANCELLED
+      existingBookedVisit.outcomeStatus = SUPERSEDED_CANCELLATION
+      existingBookedVisit.active = false
+      visitRepository.saveAndFlush(existingBookedVisit)
+      revisedVisit = true
+    }
+
+    val visitToBook = visitRepository.findReservedVisit(reference) ?: throw VisitNotFoundException("Could not find reserved visit $reference not found")
```

Comment:
This is a tricky one,... there is no direct connection between a reserved visit and the prior booked visit, in-fact there could be 2 or more  (edge case) but different call centre staff could amend the same visit at the same time. This is probably not a problem for now but will be a problem for visitor bookings!

Reply:
But a booking visitor can only amend their own visits - and no-one else can - so I think this situation is less likely to occur than with staff bookings.

But visitor users could try to corrupt the system by using multiple devices simultaneously - so we probably do need to fix this.

Reply:
>so we probably do need to fix this.
I assuming at a later date...!

Reply:
as discussed in planning this will return the latest one only